pkg/eval/json_evaluator_model.go

```diff
@@ -8,45 +8,139 @@ import (
 	"github.com/open-feature/flagd/pkg/logger"
 )
 
-type Flags struct {
-	Flags map[string]Flag `json:"flags"`
+type Flag struct {
+	State          string          `json:"state"`
+	DefaultVariant string          `json:"defaultVariant"`
+	Variants       map[string]any  `json:"variants"`
+	Targeting      json.RawMessage `json:"targeting,omitempty"`
+	Source         string          `json:"source"`
 }
 
 type Evaluators struct {
 	Evaluators map[string]json.RawMessage `json:"$evaluators"`
 }
 
-func (f Flags) Merge(logger *logger.Logger, source string, ff Flags) (Flags, map[string]interface{}) {
+type Flags struct {
+	Flags map[string]Flag `json:"flags"`
+}
+
+// Add new flags from source
+func (f Flags) Add(logger *logger.Logger, source string, ff Flags) map[string]interface{} {
+	notifications := map[string]interface{}{}
+
+	for k, newFlag := range ff.Flags {
+		if storedFlag, ok := f.Flags[k]; ok && storedFlag.Source != source {
+			logger.Warn(fmt.Sprintf(
+				"flag with key %s from source %s already exist, overriding this with flag from source %s",
+				k,
+				storedFlag.Source,
+				source,
+			))
+		}
+
+		notifications[k] = map[string]interface{}{
+			"type":   string(NotificationCreate),
+			"source": source,
+		}
+
+		// Store the new version of the flag
+		newFlag.Source = source
+		f.Flags[k] = newFlag
```

Comment:
Is there a risk of race conditions if these methods are called by multiple goroutines? Is that a possibility?

Reply:
We avoid race conditions with a mutex from Runtime [1]. So this operation is safe. 

[1] - https://github.com/open-feature/flagd/blob/main/pkg/runtime/runtime.go#L93

Reply:
It may be worth adding a comment to the `SetState` function that it's not safe to call it concurrently

Reply:
Improved comments to highlight non-thread safe fact